front/lib/tracking/amplitude/client/index.ts

```diff
@@ -145,4 +150,65 @@ export class AmplitudeClientSideTracking {
     const client = getBrowserClient();
     return client.flush().promise;
   }
+
+  static trackAssistantBuilderOpened({
+    isNew,
+    assistantName,
+    templateName,
+    workspaceId,
+  }: {
+    isNew: boolean;
+    templateName?: string;
+    assistantName?: string;
+    workspaceId: string;
+  }) {
+    const client = getBrowserClient();
+    const event = new AssistantBuilderOpened({
+      isNew,
+      templateName,
+      assistantName: assistantName || "",
+    });
+    client.track({
+      ...event,
+      groups: workspaceId
+        ? {
+            [GROUP_TYPE]: workspaceId,
+          }
+        : undefined,
+    });
+
+    return AmplitudeClientSideTracking.flush();
+  }
+
+  static trackAssistantBuilderStepViewed({
+    step,
+    isNew,
+    assistantName,
+    templateName,
+    workspaceId,
+  }: {
+    step: AssistantBuilderStepViewedProperties["assistantBuilderStep"];
+    isNew: boolean;
+    templateName?: string;
+    assistantName?: string;
+    workspaceId: string;
+  }) {
+    const client = getBrowserClient();
+    const event = new AssistantBuilderStepViewed({
+      assistantBuilderStep: step,
+      isNew,
+      templateName,
+      assistantName: assistantName || "",
```

Comment:
I would have liked to have this property be `string | undefined` but unfortunately, the "nullable" option of a property is set at the property level on Amplitude tracking plan definition, and we already have this set for the `AssistantCreated` event, where it's set to "required" :/ So we'll have to live with it I think.